tests/mssql/mssql_integration_test.go

```diff
@@ -140,4 +164,197 @@ func TestMSSQLToolEndpoints(t *testing.T) {
 	tests.RunMCPToolCallMethod(t, mcpMyFailToolWant, mcpSelect1Want)
 	tests.RunExecuteSqlToolInvokeTest(t, createTableStatement, select1Want)
 	tests.RunToolInvokeWithTemplateParameters(t, tableNameTemplateParam)
+
+	// Run specific MSSQL tool tests
+	runMSSQLListTablesTest(t, tableNameParam, tableNameAuth)
+}
+
+func runMSSQLListTablesTest(t *testing.T, tableNameParam, tableNameAuth string) {
+	// TableNameParam columns to construct want.
+	const paramTableColumns = `[
+        {"column_name": "id", "data_type": "INT", "column_ordinal_position": 1, "is_not_nullable": true},
+        {"column_name": "name", "data_type": "VARCHAR(255)", "column_ordinal_position": 2, "is_not_nullable": false}
+    ]`
+
+	// TableNameAuth columns to construct want
+	const authTableColumns = `[
+		{"column_name": "id", "data_type": "INT", "column_ordinal_position": 1, "is_not_nullable": true},
+		{"column_name": "name", "data_type": "VARCHAR(255)", "column_ordinal_position": 2, "is_not_nullable": false},
+		{"column_name": "email", "data_type": "VARCHAR(255)", "column_ordinal_position": 3, "is_not_nullable": false}
+    ]`
+
+	const (
+		// Template to construct detailed output want.
+		detailedObjectTemplate = `{
+            "schema_name": "dbo",
+            "object_name": "%[1]s",
+            "object_details": {
+                "owner": "dbo",
+                "triggers": [],
+                "columns": %[2]s,
+                "object_name": "%[1]s",
+                "object_type": "TABLE",
+                "schema_name": "dbo"
+            }
+        }`
+
+		// Template to construct simple output want
+		simpleObjectTemplate = `{"object_name":"%s", "schema_name":"dbo", "object_details":{"name":"%s"}}`
+	)
+
+	// Helper to build json for detailed want
+	getDetailedWant := func(tableName, columnJSON string) string {
+		return fmt.Sprintf(detailedObjectTemplate, tableName, columnJSON)
+	}
+
+	// Helper to build template for simple want
+	getSimpleWant := func(tableName string) string {
+		return fmt.Sprintf(simpleObjectTemplate, tableName, tableName)
+	}
+
+	invokeTcs := []struct {
+		name           string
+		api            string
+		requestBody    io.Reader
+		wantStatusCode int
+		want           string
+	}{
+		{
+			name:           "invoke list_tables detailed output",
+			api:            "http://127.0.0.1:5000/api/tool/list_tables/invoke",
+			requestBody:    bytes.NewBuffer([]byte(fmt.Sprintf(`{"table_names": "%s"}`, tableNameAuth))),
```

Comment:
```suggestion
			requestBody:    fmt.Sprintf(`{"table_names": "%s"}`, tableNameAuth),
```
Let's keep this simple, since all of the test cases require casting to bytes, we can do that at the bottom.